src/parca_agent.py

```diff
@@ -42,25 +88,26 @@ def remove(self):
         """Remove the Parca Agent snap, preserving config and data."""
         self._snap.ensure(snap.SnapState.Absent)
 
-    def configure(self, config, restart=True):
-        """Configure Parca Agent on the host system. Restart Parca Agent by default."""
-        self._snap.set({"remote-store-address": config.get("remote-store-address", "")})
-        self._snap.set({"remote-store-insecure": config.get("remote-store-insecure", "false")})
-        self._snap.set({"remote-store-bearer-token": config.get("remote-store-bearer-token", "")})
+    @property
+    def _arch(self) -> str:
+        return get_system_arch()
 
-        # Restart the snap service
-        if restart:
-            self._snap.restart()
+    @property
+    def target_revision(self) -> Optional[int]:
+        """The snap revision we want to install."""
+        return self._snap_revisions.get((self._confinement, self._arch), None)
 
     @property
-    def installed(self):
+    def installed(self) -> bool:
         """Report if the Parca Agent snap is installed."""
         return self._snap.present
 
     @property
-    def running(self):
+    def running(self) -> bool:
         """Report if the 'parca-agent-svc' snap service is running."""
-        return self._snap.services["parca-agent-svc"]["active"]
+        if self.installed:
+            return self._snap.services["parca-agent-svc"]["active"]
```

Comment:
I'm wondering if for extra safety we shouldn't wrap this up in an `except: KeyError`.